extensions/nextcloud/src/search/hooks.tsx

```diff
@@ -0,0 +1,98 @@
+import path from "path";
+import { useCallback, useState } from "react";
+import { useQuery, webdavRequest } from "../nextcloud";
+import { getPreferences } from "../preferences";
+
+export function useSearch() {
+  const [query, setQuery] = useState<string>();
+
+  const { data, isLoading } = useQuery((signal) => performSearch(signal, query), [query]);
+
+  const search = useCallback((query: string) => {
+    setQuery(query);
+  }, []);
+
+  return {
+    search,
+    isLoading,
+    results: data ?? [],
+  };
+}
+
+function makeBodyForSearch({ username, query, scope = "" }: { username: string; query: string; scope?: string }) {
+  return `<?xml version="1.0" encoding="UTF-8"?>
+<d:searchrequest xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
```

Comment:
```suggestion
<d:searchrequest xmlns:d="DAV:" xmlns:oc="https://owncloud.org/ns">
```
Shouldn't this be on a secure connection?

Reply:
They are taken from [the official API documentation](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/basic.html), and I'm not so sure about changing the XML namespace schema address as some servers may require that URL particularly be of `http://...`.